corehq/form_processor/models/forms.py

```diff
@@ -836,3 +836,18 @@ def commcare_version(self):
         version_text = get_commcare_version_from_appversion_text(self.appVersion)
         if version_text:
             return LooseVersion(version_text)
+
+
+class TempFormCache:
+    def __init__(self, domain):
+        self.domain = domain
+        self.cache = {}
+
+    def get_forms(self, form_ids):
+        cached_forms = [self.cache[form_id] for form_id in form_ids if form_id in self.cache]
+        get_forms = XFormInstance.objects.get_forms([form_id for form_id in form_ids if form_id not in self.cache],
+                                                    self.domain, ordered=True)
```

Comment:
I don't see where these forms are added to the cache. Did I miss it?

Would it be worth adding tests?

Reply:
This will hit the database with an empty list if all of the forms are already cached. It would be more efficient to skip it if there are no forms that are not cached.